optuna/integration/lightgbm.py

```diff
@@ -72,11 +72,9 @@ def __call__(self, env):
             if valid_name != target_valid_name or metric != self.metric:
                 continue
 
-            # TODO(ohta): Deal with maximize direction
             if is_higher_better:
-                raise ValueError(
-                    'Pruning using metrics to be maximized has not been supported yet '
-                    '(validation_name: {}, metric: {}).'.format(valid_name, metric))
+                assert self.trial.storage.get_study_direction(self.trial.study_id) == \
```

Comment:
I think an `else` block like the following is needed for checking the consistency when `is_higher_better == False`:
```python
else:
    assert self.trial.storage.get_study_direction(self.trial.study_id) == optuna.structs.StudyDirection.MINIMIZE
```
```

Reply:
You're right. I fixed it.
Thank you for pointing it out.

Reply:
Since this is user facing when the assertion fails, we might want to clarify the reason with an error message.

```python
else:
    if self.trial.storage.get_study_direction(self.trial.study_id) != optuna.structs.StudyDirection.MINIMIZE:
        raise ValueError(...)
```

Reply:
I agree with you. I added the following error message. Thank you. 

> 'The intermediate values are inconsistent with the objective values in terms of study directions. Please specify a metric to be minimized for LightGBMPruningCallback.'